### PR TITLE
feat(otelcol/exporter/otlp*): resync drifted defaults with upstream

### LINKED ISSUE #6467: Resync defaults for otelcol.exporter.otlp with upstream
This file notes that our defaults have become desynced with upstream:

https://github.com/grafana/agent/blob/744a4b7723eb4068d65e289a3ba709408868cede/converter/internal/otelcolconvert/testdata/otlp.yaml#L9-L14

Where possible, we should resync. If resyncing a particular field would be classified as a breaking change, we should be careful and consider whether resyncing is worth it.

### LINKED ISSUE #6485: Resync defaults for otelcol.exporter.otlphttp with upstream
#6484 found that our defaults for otelcol.exporter.otlphttp have desynced with upstream (similar to #6467 for otelcol.exporter.otlp). 

Where possible, we should resync. If resyncing a particular field would be classified as a breaking change, we should be careful and consider whether resyncing is worth it.

### PR DESCRIPTION
**Pull request by hainenber**
#### Which issue(s) this PR fixes

Fixes #6485
Fixes #6467 

#### PR Checklist

- [x] CHANGELOG.md updated
- [x] Documentation added

### CODE REVIEW

File: docs/sources/shared/flow/reference/components/otelcol-queue-block.md [hunk 1]
```diff
@@ -16,14 +16,14 @@ Name            | Type      | Description
...
+`queue_size`    | `number`  | Maximum number of unwritten batches allowed in the queue at the same time. | `1000`  | no
 
 When `enabled` is `true`, data is first written to an in-memory buffer before sending it to the configured server.
 Batches sent to the component's `input` exported field are added to the buffer as long as the number of unsent batches doesn't exceed the configured `queue_size`.
 
 `queue_size` determines how long an endpoint outage is tolerated.
-Assuming 100 requests/second, the default queue size `5000` provides about 50 seconds of outage tolerance.
-To calculate the correct value for `queue_size`, multiply the average number of outgoing requests per second by the time in seconds that outages are tolerated.
+Assuming 100 requests/second, the default queue size `1000` provides about 10 seconds of outage tolerance.
+To calculate the correct value for `queue_size`, multiply the average number of outgoing requests per second by the time in seconds that outages are tolerated. Note that a very high figure can cause unexpected Out Of Memory (OOM) kills.
```
Comment (clayton-cornell) [inline docs/sources/shared/flow/reference/components/otelcol-queue-block.md:26]:
```suggestion
To calculate the correct value for `queue_size`, multiply the average number of outgoing requests per second by the time in seconds that outages are tolerated. A very high value can cause Out Of Memory (OOM) kills.
```

**Review by clayton-cornell — COMMENTED** [review]
One suggestion for consistency and simplifying.

**Comment by clayton-cornell** [comment]
Over to @grafana/grafana-agent-maintainers for code review

**Review by clayton-cornell — DISMISSED** [review]

**Comment by hainenber** [comment]
Looks like I need to update some defaults as well. I'm at work atm, would troubleshoot and remediate later

File: internal/component/otelcol/config_grpc.go [hunk 2]
```diff
@@ -13,6 +13,8 @@ import (
 	otelextension "go.opentelemetry.io/collector/extension"
 )
 
+const DEFAULT_BALANCER_NAME = "pick_first"
```
Comment (rfratto) [inline internal/component/otelcol/config_grpc.go:16]: Idiomatically, it's unusual for constants in Go to be SNAKE_CASE like this. Can this be called `DefaultBalancerName` instead?

**Review by rfratto — COMMENTED** [review]
Thanks! Just one comment about the constant name, then this should be good to go :)

File: internal/component/otelcol/config_grpc.go (hunk 2 above)
Comment (hainenber) [reply internal/component/otelcol/config_grpc.go]: Done :D

File: docs/sources/shared/flow/reference/components/otelcol-queue-block.md (hunk 1 above)
Comment (clayton-cornell) [reply docs/sources/shared/flow/reference/components/otelcol-queue-block.md:26]: I don't see this suggestion applied. Generally we do not want to state `Note that` in the documentation. If it's a note, then we should make it a note... In this case a factual statement is enough.

Comment (hainenber) [reply docs/sources/shared/flow/reference/components/otelcol-queue-block.md:26]: My bad, I failed to pull the changes here to my local. Commited!

**Review by clayton-cornell — APPROVED** [review]
Docs look good. Over to you @rfratto for final check :-)

**Review by rfratto — APPROVED** [review]
Thank you!

**Pull request merged**